src/views/clusteroverview/SettingsTab/ClusterTab/components/GeneralSettings/MemoryDensity/MemoryDensity.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { FC, useCallback } from 'react';
+
+import HyperConvergedModel from '@kubevirt-ui/kubevirt-api/console/models/HyperConvergedModel';
+import SectionWithSwitch from '@kubevirt-utils/components/SectionWithSwitch/SectionWithSwitch';
+import { HyperConverged } from '@kubevirt-utils/hooks/useHyperConvergeConfiguration';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { k8sPatch } from '@openshift-console/dynamic-plugin-sdk';
+import { Skeleton } from '@patternfly/react-core';
+
+import { useIsAdmin } from '../../../../../../../utils/hooks/useIsAdmin';
+import ExpandSection from '../../../../ExpandSection/ExpandSection';
+
+type MemoryDensityProps = {
+  hyperConvergeConfiguration: [hyperConvergeConfig: HyperConverged, loaded: boolean, error: any];
+  newBadge: boolean;
+};
+
+const MemoryDensity: FC<MemoryDensityProps> = ({ hyperConvergeConfiguration, newBadge }) => {
+  const { t } = useKubevirtTranslation();
+  const isAdmin = useIsAdmin();
+  const [hyperConverge, hyperLoaded] = hyperConvergeConfiguration;
+
+  const onChange = useCallback(
+    (checked: boolean) => {
+      k8sPatch({
+        data: [
+          {
+            op: 'replace',
+            path: `/spec/featureGates/enableHigherDensityWithSwap`,
+            value: checked,
+          },
+          {
+            op: 'replace',
+            path: `/spec/higherWorkloadDensity/memoryOvercommitPercentage`,
+            value: checked ? 150 : 100,
+          },
+        ],
+        model: HyperConvergedModel,
+        resource: hyperConverge,
+      });
+    },
+    [hyperConverge],
+  );
+
+  if (!hyperLoaded) return <Skeleton width={'300px'} />;
+
+  const higherDensity = hyperConverge.spec?.featureGates?.enableHigherDensityWithSwap;
+
+  return (
+    <ExpandSection toggleText={t('Memory density')}>
+      <SectionWithSwitch
+        helpTextIconContent={t('Configures the VM workloads to use swap for higher density')}
+        id="memory-density-feature"
+        isDisabled={!hyperLoaded || !isAdmin || higherDensity === undefined}
```

Comment:
what do you think about changing last condition to isEmpty(higherDensity) in case of other nullish value?

Reply:
sure